oak_attestation_verification/src/verifier.rs

```diff
@@ -294,51 +324,69 @@ fn verify_root_layer_endorsements(
         .as_ref()
         .ok_or_else(|| anyhow::anyhow!("no stage0 reference value"))?;
 
+    // WIP: Parse attestation report and verify stage0 measurement.
     verify_transparent_release_endorsement(now_utc_millis, e_stage0, r_stage0)
 }
 
 /// Verifies all ingredients of the kernel layer, which is common to both
 /// Oak Restricted Kernel and Oak Containers setups.
-fn verify_kernel_layer_endorsements(
+fn verify_kernel_layer(
     now_utc_millis: i64,
+    l: &LayerEvidence,
     e: &KernelLayerEndorsements,
     r: &KernelLayerReferenceValues,
 ) -> anyhow::Result<()> {
+    let claims = claims_set_from_evidence_layer(l)?;
+
     let e_kernel_image = e
         .kernel_image
         .as_ref()
         .ok_or_else(|| anyhow::anyhow!("no kernel endorsement"))?;
-
     let r_kernel_image = r
         .kernel_image
         .as_ref()
         .ok_or_else(|| anyhow::anyhow!("no kernel reference value"))?;
 
-    verify_transparent_release_endorsement(now_utc_millis, e_kernel_image, r_kernel_image)?;
+    verify_item(
```

Comment:
We have had requests from customer teams to split extracting the evidence from comparing it with the reference values and endorsemets to allow them to implement different verification approaches.

Concretely this means that we should define a struct/proto for the various layer measurements, and extracting the measurements into that struct/proto as a separate step, and then just passing the measurement from that struct into this functions.

For me the most logical approach would be to combine validating the DICE certificate chain with extracting the measurements as part of `verify_dice_chain` and extending the `DiceChainResult` struct to include the measurements.

This can be done as a separate refactoring step in a follow-up PR.